MvvmCross/Platforms/Uap/Views/MvxWindowsPage.cs

```diff
@@ -68,12 +70,17 @@ public IMvxViewModel ViewModel
 
         public void ClearBackStack()
         {
-            throw new NotImplementedException();
-            /*
-            // note - we do *not* use CanGoBack here - as that seems to always returns true!
-            while (NavigationService.BackStack.Any())
-                NavigationService.RemoveBackEntry();
-         */
+            var backStack = base.Frame?.BackStack;
+
+            if (backStack != null)
+            {
+                while (backStack.Any())
+                {
+                    backStack.RemoveAt(0);
+                }
+            }
+
+            SystemNavigationManager.GetForCurrentView().AppViewBackButtonVisibility = AppViewBackButtonVisibility.Collapsed;
```

Comment:
Please extract into a separate method eg UpdateBackButtonVisibility that is protected virtual, allowing it to be overridden and intercepted.
I would suggest making ClearBackStack virtual too.